source/codegen/templates/proto_helpers.mako

```diff
@@ -118,7 +118,7 @@ message ${common_helpers.snake_to_pascal(function)}Response {
 message ${custom_type["grpc_name"]} {
 % for field in common_helpers.filter_parameters_for_grpc_fields(custom_type["fields"]):
 <%
-  field_type = field.get("enum", None) or common_helpers.get_grpc_type(field["type"], config)
+  field_type = field.get("enum", None) or field.get("grpc_type", None) or common_helpers.get_grpc_type(field["type"], config)
```

Comment:
This is a pretty weird line. Is the issue here that `"grpc_type`" is set on all function params as a metadata mutation (or in source), but there's not an equivalent for custom type fields? So we have semi-reasonable code for accessing param types as `param["grpc_type"]` but can't do the same for fields here?

Would it be reasonable to change `metadata_mutation` to do the same for custom type fields?

If not, this is fine as a one off. It could make sense to add something to `common_helpers`, but there are already a lot of `get_grpc_whatever` methods in there. So I think that would make it worse.